torchtune/datasets/_instruct_raft.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Any, Callable, Dict, List, Mapping, Optional, Tuple
+
+import numpy as np
+from torch.utils.data import Dataset
+from torchtune.config._utils import _get_instruct_template
+
+from torchtune.data import (
+    CROSS_ENTROPY_IGNORE_IDX,
+    InstructTemplate,
+    Message,
+    validate_messages,
+)
+from torchtune.datasets._utils import load_deep_lake_dataset
+
+from torchtune.modules.tokenizers import Tokenizer
+
+
+class InstructDatasetDeepLakeRAFT(Dataset):
```

Comment:
High-level question on this class: is the only difference between this and `InstructDataset` the usage of `load_deep_lake_dataset` instead of `load_dataset`? If so, I wonder if this is something we should consider parametrizing rather than having to write an entirely new dataset class. cc @RdoubleA for any thoughts here

Reply:
yes I think it can be parameterized, can you give me some advice on how we can proceed here?